sumo-api/sumo/base/colored_category.pyx

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# Copyright 2004-present Facebook. All Rights Reserved.
+
+import csv
+import numpy as np
+cimport numpy as np
+import os
+
+
+class ColoredCategory(object):
+    """ Class to generate distinct RGB values for different categories of a dataset."""
+
+    def __init__(self, csv_file):
+        """ Create an object that holds color information of categories in
+        <dataset_name> dataset.
+
+        Inputs:
+        csv_file (str) - filepath to an external csv file containing color mapping
+            information.
+
+        Exceptions:
+        IOError - raised if csv_file does not exist.
+
+        Note:
+        The CSV file must be in the the following format:
+        id,category,is_evaluated,color
+
+        example:
+        id,category,is_evaluated,color
+        0,empty,False,(152, 245, 255)
+        1,accordion,False,(30, 144, 255)
+        2,air_conditioner,True,(122, 197, 205)
+        3,amplifier,False,(110, 123, 139)
+        ...
+        """
+        if not os.path.exists(csv_file):
+            raise IOError("File {} does not exist.".format(csv_file))
+
+        with open(csv_file, "r", newline="") as f:
+            reader = csv.DictReader(f)
+            self._data = [{
+                "category": line["category"],
+                "is_evaluated": line["is_evaluated"],
+                "color": tuple(map(int, line["color"].split(";")))} for line in reader]
+
+        self._reverse_data = {
+            self._data[id]["category"]: self._data[id]["color"] for id, x in enumerate(self._data)
+        }
+        cdef np.ndarray lut = np.zeros((0))
+        self._lut = lut
+
+    @property
+    def LUT(self):
+        """The lookup table for conversion of indexes to RGB values."""
+        if self._lut.shape[0] == 0:
+            total = len(self._data)
+            self._lut = np.array([x["color"]
+                for x in self._data], dtype=np.uint8)
+        return self._lut
+
+    def from_category_id_to_rgb(self, category_id):
+        """ Convert a category id to its corresponding color.
+        Inputs:
+        category_id (int) - id of the category
+
+        Returns
+        An RGB tuple, which corresponds to the category_id
```

Comment:
Specify range and type of the tuple elements.  Are they 0-255 ints, or 0-1 floats?